pkg/api/handlers/compat/containers.go

```diff
@@ -44,11 +45,17 @@ func RemoveContainer(w http.ResponseWriter, r *http.Request) {
 	runtime := r.Context().Value("runtime").(*libpod.Runtime)
 	name := utils.GetName(r)
 	con, err := runtime.LookupContainer(name)
-	if err != nil {
-		utils.ContainerNotFound(w, name, err)
-		return
+	if err != nil && errors.Cause(err) == define.ErrNoSuchCtr {
+		if _, err := runtime.EvictContainer(r.Context(), name, query.Vols); err != nil {
+			if errors.Cause(err) == define.ErrNoSuchCtr {
+				logrus.Debugf("Ignoring error (--allow-missing): %q", err)
+				w.WriteHeader(http.StatusNoContent)
+			} else {
+				utils.InternalServerError(w, err)
+			}
+			return
+		}
```

Comment:
Ah, there's a fallthrough here when `err != nil` and the newly-added `Cause` check fails.